### DISCUSSION TITLE
Set light group transition speed

### DISCUSSION THREAD

**Discussion by stuartmaxwell** [Koenkk/zigbee2mqtt#12859] [General]
Hi all. I'm running Zigbee2MQTT in home assistant with a number of Hue bulbs. I have a couple of light groups created in Zigbee2MQTT, but I can't see any way of configuring the transition speed from the frontend. I tried to add the transition setting to the groups.yaml file that is created, but this doesn't seem to do anything and gets removed when I make any changes with the frontend.

Am I correct in thinking that I can't configure the transition speed of a light group from the frontend?

**Comment by giulianofichera** [comment]
I'm having the same problem, I added `transition: 5` in every way possible:
- In devices.yaml as a [default value](https://www.zigbee2mqtt.io/guide/configuration/devices-groups.html#specific-device-options):
```
device_options:  
  transition: 5
```
- In devices.yaml as a single light option:
```
'0x0017880103fd3b4f':  
    friendly_name: Roof 2   
    transition: 5
```
    
- And in configuration.yaml:
`  "device_options": {"transition": 5},`

Yet when I examine the message through MQTT Explorer, there is no transition argument:
`{"state": "ON", "brightness": 254}`

The light in question is a hue bulb that [supports transition](https://www.zigbee2mqtt.io/devices/9290012573A.html#philips-9290012573a), as when I manually send `{"state": "ON", "brightness": 254,"transition":5}` the bulb transitions to the new brightness.

Zigbee2MQTT is missing the transition argument when sending the message.

**Comment by ianclegg** [comment]
@stuartmaxwell @giulianofichera  did you get this to work?
I have set the `transition` property in the z2m `configuration.yaml` as documented here:
https://www.zigbee2mqtt.io/guide/configuration/devices-groups.html#groups

But it does not appear to work for me either. I have set it on the device and the group. However, transitions do work when I set the configuration on an individual device through `Settings (Specific)` in the UI and then set the state via the UI - but that appears to work in a different way, it appears to configure the device in such a way that the device stores the transition to use - because when I toggle state I don't see the `transition` property in the payload (though I can see the device is using the setting that was configured)

In fact, I see the same as @giulianofichera. I never see `transition` being published by z2m in MQTT Explorer, despite it being set in the for the devices and the group they deploy to in the `configuration` yaml?

**Reply by giulianofichera** [reply]
Hi @ianclegg my issue was resolved by switching to the [new repo](https://github.com/zigbee2mqtt/hassio-zigbee2mqtt) and updating to the last version. The old repo had a very old version of Zigbee2MQTT that didn't support transitions.

Now, in configuration.yaml in the Zigbee2MQTT folder I have set up the following:
`device_options:
  transition: 0.3`
  
  Everything works correctly now. Let me know if this solves you problem!

**Reply by ianclegg** [reply]
ahhh, let me give that try

**Reply by ianclegg** [reply]
@giulianofichera thanks! I have the latest version, and the `device_options` the transition for groups works great! 🍾 

I am still seeing an issue with z2m **scenes**, I imagine this might be implemented differently to groups and perhaps configured somewhere else. 

The behaviour I have is that transition applies when turning the group off or on, for example but publishing `{ "state": "OFF" }` or `{ "state": "ON" }` or setting brightness, but when I create and recall a **scene** in z2m the transition does not apply.

Are you using z2m scenes at all, if so, do you see this behaviour too?

**Reply by ianclegg** [reply]
@giulianofichera to be honest i'm so pleased that groups work, I might leave scenes for another day :)

**Reply by giulianofichera** [reply]
@ianclegg that's great to hear! Sadly I don't use Home Assistant scenes so I can't help with that, but I do use scenes with [diyHue](https://github.com/maxbec/hassio-diyHue) and [Hue Essentials](https://www.hueessentials.com/) so everything works exactly the same as when I had my Hue hub. The only downside is that you can't call scenes from HA, but I really don't have any use for that at the moment.  

Regardless of what you want to do, you might want to use that app with your phone. Let me know if there's anything else I can help with.

Cheers!

**Reply by MaJerle** [reply]
In 2024 July, I have this in my `zigbee2mqtt/configuration.yaml` file
```yaml
device_options:
  legacy: false
  transition: 0.5
```

but when controlling the brightness for a Z2M lights group from HA dashboard, changes are instant, without transition. If I control brightness of each light entity separately, I can see the transition nicely.

**Reply by sindreruud** [reply]
Same for me @MaJerle , did you find a solution to this?

**Reply by MaJerle** [reply]
> Same for me @MaJerle , did you find a solution to this?

In the *Z2M* configuration.yaml file, I put the `transition: 1` to every device. Same to groups. Like so:


```yaml
device_options:
  legacy: false
  transition: 1
devices:
  '0x00178000longhexnumberid':
    friendly_name: Friendly name goes here
    transition: 1
groups:
  '1':
    friendly_name: lights_mygroupname
    transition: 1
    devices:
      - 0x0017000longhexnumberid
```

So `device_options`, `devices` and `groups` all contain `transition: 1` and now it works. I don't know which, but one of them made it work.

**Reply by myevit** [reply]
> > Same for me @MaJerle , did you find a solution to this?
> 
> In the _Z2M_ configuration.yaml file, I put the `transition: 1` to every device. Same to groups. Like so:
> 
> ```yaml
> device_options:
>   legacy: false
>   transition: 1
> devices:
>   '0x00178000longhexnumberid':
>     friendly_name: Friendly name goes here
>     transition: 1
> groups:
>   '1':
>     friendly_name: lights_mygroupname
>     transition: 1
>     devices:
>       - 0x0017000longhexnumberid
> ```
> 
> So `device_options`, `devices` and `groups` all contain `transition: 1` and now it works. I don't know which, but one of them made it work.

for me device_options and groups were enough.

**Comment by icheered** [comment]
If someone finds this in the future and has the issue that after reloading zigbee2mqtt the `configuration.yaml` file is reverted so the option isn't saved, you can do the following:
1. In the zigbee2mqtt plugin, go to settings => dev console
2. Publish to the topic: `zigbee2mqtt/bridge/request/group/options`
3. Publish the payload: `{"id": "<YOUR FRIENDLY GROUP NAME HERE>", "options": {"transition": 1}}`

Then click send, and it should have correctly updated the  'configuration.yaml`. Now controlling the group through HA should be smooth :)
